iroh-sync/src/ranger.rs

```diff
@@ -629,11 +491,143 @@ where
 #[cfg(test)]
 mod tests {
     use proptest::prelude::*;
-    use std::{cell::RefCell, fmt::Debug, rc::Rc};
+    use std::{cell::RefCell, collections::BTreeMap, convert::Infallible, fmt::Debug, rc::Rc};
     use test_strategy::proptest;
 
     use super::*;
 
+    #[derive(Debug)]
+    pub struct SimpleStore<K, V> {
+        data: BTreeMap<K, V>,
+    }
+
+    impl<K, V> Default for SimpleStore<K, V> {
+        fn default() -> Self {
+            SimpleStore {
+                data: BTreeMap::default(),
+            }
+        }
+    }
+
+    impl<K, V> RangeEntry for (K, V)
+    where
+        K: RangeKey + PartialEq + Clone + Default + Debug,
+        V: Debug + Clone + PartialOrd,
+    {
+        type Key = K;
+
+        fn key(&self) -> &Self::Key {
+            &self.0
+        }
+
+        fn as_fingerprint(&self) -> Fingerprint {
+            let mut hasher = blake3::Hasher::new();
+            hasher.update(format!("{:?}", self.0).as_bytes());
```

Comment:
this should include the value as well probably, to be more consistent in testing this